userpage/views.py

```diff
@@ -11,15 +14,62 @@ def validate_user(self):
         input: self.input['student_id'] and self.input['password']
         raise: ValidateError when validating failed
         """
-        raise NotImplementedError('You should implement UserBind.validate_user method')
+        if len(self.input['student_id']) > 32:
+            raise ValidateError('Invalid student id')
 
     def get(self):
         self.check_input('openid')
-        return User.get_by_openid(self.input['openid']).student_id
+        try:
+            studentId = User.get_by_openid(self.input['openid']).student_id
+        except:
+            studentId = ''
+        return studentId
 
     def post(self):
         self.check_input('openid', 'student_id', 'password')
         user = User.get_by_openid(self.input['openid'])
         self.validate_user()
         user.student_id = self.input['student_id']
         user.save()
```

Comment:
because I am not familiar with 'userpage', I think in this section lock is needed.